src/IPC/PipeTx.cc

```diff
@@ -1,21 +1,75 @@
 #include "IPC.h"
 
-void PipeTx::send(void) {
-    this->setupPipeTx();
+#include <sys/stat.h>   /* mkfifo() */
+#include <fcntl.h>      /* O_CREAT, O_WRONLY */
+#include <iostream>     /* std::cout std::ios seekg() */
+#include <unistd.h>     /* write() */
 
-    this->fileSizeTx();
+#define FIFO "./fifoChannel"
 
+void PipeTx::send(void) {
+    this->setupPipeTx();
+    
     this->pipeTx();
 }
 
 void PipeTx::setupPipeTx(void) {
-    ;
-}
-
-void PipeTx::fileSizeTx(void) {
-    ;
+    mkfifo(FIFO, 0666);                         /* read/write for user/group/others */
+    this->fd = open(FIFO, O_CREAT | O_WRONLY);  
```

Comment:
fifo should be alredy created by mkfifo, if not O_CREAT will just create regular file not pipe